### DISCUSSION TITLE
Improve discoverability of changes by package

### DISCUSSION THREAD

**Discussion by dominikg** [nextauthjs/next-auth#8926] [Ideas]
### What is the improvement or update you wish to see?

This repository hosts a lot of packages and so the github releases page requires searching to find information about a specific package. This can get tricky eg if you search for `@authjs/core` https://github.com/nextauthjs/next-auth/releases?q=%22%40auth%2Fcore%22&expanded=true  where it finds a lot of releases of packages that depend on it, drowning out the actual releases.

It would be great to have CHANGELOG.md files in addition in each package so it's easier to discover what changed between versions without having to resort to advanced search queries in gh releases or looking at the git history directly

note: a release entry for 0.17.0 exists: https://github.com/nextauthjs/next-auth/releases/tag/%40auth%2Fcore%400.17.0
This is mostly about how hard it is to find and that it's not easy to see all releases of core on one page.

### Is there any context that might help us understand?

came here to research what changed in v 0.17.0 after `pnpm outdated` showed a new version available.

**Comment by JanJakes** [comment]
Just came across this.

A search like this works for core v0: https://github.com/nextauthjs/next-auth/releases?q=%40auth%2Fcore%400
Similarly this one for next-auth v4: https://github.com/nextauthjs/next-auth/releases?q=next-auth%404

But I agree it would be nice to have a better changelog.

**Comment by TorbjornHoltmon** [comment]
Changelogs are sorely needed!

It is so hard and frustrating to find and page through all the dependencies to find the changelog I am looking for!

When going from version X to Y I basically cannot get a nice overview of what changes have been made without the hacky search that JanJakes provided.

Adopting changeset would probably be the best, and writing a script to get all the old changes into each package.

I am not sure whats used today, looks like it might be a custom script.
